src/cpp/src/tokenizer.cpp

```diff
@@ -117,184 +114,108 @@ class Tokenizer::TokenizerImpl {
 
     TokenizerImpl() = default;
 
-    TokenizerImpl(std::filesystem::path tokenizer_path, const ov::AnyMap& properties)
-        : m_chat_template{chat_template_from_tokenizer_json_if_exists(tokenizer_path)} {
-        ov::Core core;
-
-        OPENVINO_ASSERT(tokenizer_path.extension() != ".xml", "'tokenizer_path' parameter should be a path to a dir not a xml file");
-
+    std::shared_ptr<ov::Core> get_core() {
+        if (m_core) {
+            return m_core;
+        }
+        m_core = std::make_shared<ov::Core>();
```

Comment:
can we use https://github.com/openvinotoolkit/openvino.genai/blob/master/src/cpp/src/utils.hpp#L87 ?